### PR TITLE
Prac 07 review code request

### PR DESCRIPTION
**Pull request by TinNilarSoe**
@DianaBukeikhanova 
@ChristineYamin
Please review my prac_07. Thank you.

### CODE REVIEW

File: prac_07/project_management.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+def save_projects(projects, filename):
+    """Save projects to the file"""
+    with open(filename, "w") as out_file:
+        out_file.write("Name	Start Date	Priority	Cost Estimate	Completion Percentage\n")
+        for project in projects:
+            out_file.write(f"{project.name}\t{project.start_date.strftime("%d/%m/%Y")}\t{project.priority}"
+                           f"\t{project.estimate_cost:.2f}\t{project.completion}\n")
+
+
+main()
```
Comment (ChristineYamin) [inline prac_07/project_management.py:151]: Perfect coding 💯

File: prac_07/myguitars.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+def main():
+    guitars = read_guitar_file(FILENAME)
+    display_sorted_guitars(guitars)
+
+
+def display_sorted_guitars(guitars):
+    """Sort and display the guitars"""
+    guitars.sort().__lt__(1)
```
Comment (ChristineYamin) [inline prac_07/myguitars.py:22]: Nice usage of  __lt__ method

File: prac_07/myguitars.py [hunk 3]
```diff
@@ -27,4 +37,15 @@ def read_guitar_file(filename):
     return guitars
 
 
+def get_information(guitars):
+    """Get guitar information and put it in a list"""
+    name = input("Name: ").title()
+    while name != "":
+        year = int(input("Year: "))
+        cost = float(input("Cost: $"))
+        print(f"{name} ({year}) : ${cost:,.2f} added.")
```
Comment (ChristineYamin) [inline prac_07/myguitars.py:46]: It is good coding because you add error checking

File: prac_07/programming_language.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    languages = [ruby, python, visual_basic]
+    print(python)
+
+    print("The dynamically typed languages are:")
+    for language in languages:
+        if language.is_dynamic():
+            print(language.name)
+
+
```
Comment (DianaBukeikhanova) [inline prac_07/programming_language.py:41]: Good codes

File: prac_07/guitar.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        """Return a string for guitar name, year and cost"""
+        return "{} ({}) : ${}".format(self.name, self.year, self.cost)
+
+    def get_age(self, current_year=2024):
+        """Return the age of the guitar"""
+        return current_year - self.year
+
+    def is_vintage(self):
+        """Return True if guitar is older than 50"""
+        if self.get_age() < 50:
```
Comment (DianaBukeikhanova) [inline prac_07/guitar.py:28]: You can use constants.

File: prac_07/myguitars.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    guitars = []
+    in_file = open(filename, "r")
+    in_file.readline()
+    for line in in_file:
+        parts = line.strip().split(",")
+        guitar = Guitar(parts[0], int(parts[1]), float(parts[2]))
+        guitars.append(guitar)
+    in_file.close()
+    return guitars
+
```
Comment (DianaBukeikhanova) [inline prac_07/myguitars.py:38]: Working code. Good job!

File: prac_07/project_management.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    start_date = input("Start date (dd/mm/yy): ")
+    while "/" not in start_date:
+        print("Please write date in format (dd/mm/yy)")
+        start_date = input("Start date (dd/mm/yy): ")
+    priority = validate_input_number("Priority: ")
+    estimate_cost = validate_input_number("Cost estimate: $")
+    completed_percentage = validate_input_number("Percent complete: ")
+    projects.append(Project(project_name, start_date, priority, estimate_cost, completed_percentage))
+    return projects
+
```
Comment (DianaBukeikhanova) [inline prac_07/project_management.py:106]: Good code

File: prac_07/guitar.py (hunk 5 above)
Comment (TinNilarSoe) [reply prac_07/guitar.py:28]: Noted.

File: prac_07/project_management.py (hunk 1 above)
Comment (TinNilarSoe) [reply prac_07/project_management.py:151]: Thank you.

File: prac_07/project_management.py (hunk 7 above)
Comment (TinNilarSoe) [reply prac_07/project_management.py:106]: Thank you.

**Pull request merged**